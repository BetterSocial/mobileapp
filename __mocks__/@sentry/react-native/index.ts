export const addBreadcrumb = jest.fn();
