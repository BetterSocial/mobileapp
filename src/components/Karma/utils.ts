export const renderCircleColor = (karma: number) => {
  switch (true) {
    case karma > 87.5:
      return ['#50BA57', '#C7EECA', '#50BA57'];
    case karma > 75:
      return ['#70BF21', '#B3E57F', '#70BF21'];
    case karma > 62.5:
      return ['#A0C527', '#D1EA82', '#A0C527'];
    case karma > 50:
      return ['#D3D20C', '#F3F370', '#D3D20C'];
    case karma > 37.5:
      return ['#E4B92E', '#FFE591', '#E4B92E'];
    case karma > 25:
      return ['#F4833C', '#F9BE9A', '#F4833C'];
    case karma > 12.5:
    default:
      return ['#DF001F', '#F5B0B9', '#DF001F'];
  }
};
