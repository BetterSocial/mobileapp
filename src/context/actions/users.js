import {SET_DATA_HUMAN_ID, SET_DATA_IMAGE, SET_DATA_USERNAME, SET_LOCAL_COMUNITY} from '../Types';

export const setDataHumenId = async (data, dispatch) => {
  let {appUserId, countryCode} = data;
  dispatch({
    type: SET_DATA_HUMAN_ID,
    payload: {
      userId: appUserId,
      countryCode: countryCode,
    },
  });
};
export const setImage = (base64, dispatch) => {
  dispatch({
    type: SET_DATA_IMAGE,
    payload: base64,
  });
};
export const setUsername = (username, dispatch) => {
  dispatch({
    type: SET_DATA_USERNAME,
    payload: username,
  });
};
export const setLocalCommunity = (location, dispatch) => {
  console.log('isi location user ', location)
  dispatch({
    type: SET_LOCAL_COMUNITY,
    payload: location,
  });
};
