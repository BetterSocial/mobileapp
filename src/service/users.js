import crashlytics from '@react-native-firebase/crashlytics';

import StorageUtils from '../utils/storage';
import api from './config';

export const verifyUser = async (userId) => {
  try {
    const resApi = await api.post('/users/verify-user', {
      user_id: userId
    });
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};

export const demoVerifyUser = async (user_id, password) => {
  try {
    const resApi = await api.post('/users/demo-verify-user-v2', {
      user_id,
      password
    });

    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    if (error?.response?.data?.status?.toLowerCase() === 'password is invalid') {
      throw new Error('Password is invalid');
    }

    return error?.response?.data;
  }
};

/**
 *
 * @param {String} query
 * @returns {import('../../types/service/UserService.typedef').UserSearchChatApiResponse | Boolean}
 */
export const searchChatUsers = async (query) => {
  try {
    const resApi = await api.get(`/users/chat/search?q=${query}`);
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return false;
  }
};

export const verifyToken = async (token) => {
  try {
    const resApi = await api.post('/users/veryfy-token', {
      token
    });
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};
export const verifyUsername = async (username) => {
  try {
    const resApi = await api.post('/users/check-username', {
      username
    });
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};
export const registerUser = async (data) => {
  try {
    const resApi = await api.post('/users/register-v2-without-upload-photo', {
      data
    });
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};

const verifyAccessToken = async () =>
  api
    .get('/users/veryfy-token-getstream', {})
    .then((res) => res.data)
    .catch((err) => {
      const code = err.response.status;
      return code;
    });

export const refreshToken = async () => {
  const token = StorageUtils.refreshToken.get();
  const options = {
    headers: {
      Authorization: `Bearer ${token}`
    }
  };

  // const resp = await fetchWithTimeout(this.url, options, 10000);
  try {
    const resp = await api.get('/users/refresh-token', options);
    return resp.data;
  } catch (err) {
    return err?.response?.data;
  }
};

export const verifyTokenGetstream = async () => {
  const status = await verifyAccessToken();
  if (status === 401) {
    const res = await refreshToken();

    if (res.code === 200) {
      return res.data.token;
    }
    return null;
  }
  return status.data;
};

export const userPopulate = async (isAnon) => {
  try {
    const resApi = await api.get(`/users/populate${isAnon ? '?allow_anon_dm=true' : ''}`);
    return resApi.data.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};

export const getBlockedUserList = async () => {
  try {
    const getBlockList = await api.get('/profiles/block');
    return getBlockList.data;
  } catch (e) {
    crashlytics().recordError(new Error(e));
    return e.response.data;
  }
};

export const deleteAccount = async () => {
  try {
    const result = await api.post('/users/delete');
    return result.data;
  } catch (e) {
    crashlytics().recordError(new Error(e));
    return null;
  }
};

export const fcmTokenService = async (body) => {
  try {
    const result = await api.post('/users/fcmtoken', body);
    return result.data;
  } catch (e) {
    crashlytics().recordError(new Error(e));
    return null;
  }
};

export const removeFcmToken = async (fcmToken) => {
  try {
    const result = await api.delete(`/users/fcmtoken-v2?fcm_token=${fcmToken}`);
    return result.data;
  } catch (e) {
    crashlytics().recordError(new Error(e));
    return null;
  }
};

export const verifyHumanIdExchangeToken = async (exchangeToken, dummy = false) => {
  const data = {
    token: exchangeToken
  };

  const exchangeTokenUrl = dummy
    ? '/users/check-exchange-token-dummy'
    : '/users/check-exchange-token';

  try {
    const result = await api.post(exchangeTokenUrl, data);

    return Promise.resolve(result?.data);
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return Promise.reject(error?.response?.data);
  }
};

export const checkPasswordForDemoLogin = async (password) => {
  try {
    const payload = {password};
    const result = await api.post('/users/password-verify-user', payload);
    if (result?.data?.code === 200) {
      return {
        success: true,
        message: result?.data?.message
      };
    }

    return {
      code: result?.data?.code,
      success: false,
      message: result?.data?.message
    };
  } catch (e) {
    crashlytics().recordError(new Error(e));
    return {
      success: false,
      code: e?.response?.data?.code,
      message: e?.response?.data?.message
    };
  }
};

export const checkFollowStatusBatch = async (targetUserIds = [], axiosOptions = {}) => {
  if (targetUserIds.length === 0) return Promise.reject(new Error('User IDs is empty'));

  try {
    const payload = {
      targetUserIds
    };

    const result = await api.post('/users/check-follow-batch', payload, axiosOptions);
    if (result?.status === 200) {
      return Promise.resolve(result?.data?.data);
    }

    return Promise.reject(result?.data?.message);
  } catch (e) {
    crashlytics().recordError(new Error(e));
    return Promise.reject(
      e?.response?.data?.message || "Can't check follow status, please try again later"
    );
  }
};
