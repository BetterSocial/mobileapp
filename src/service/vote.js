import crashlytics from '@react-native-firebase/crashlytics';
import api from './config';

export const upVote = async (data) => {
  try {
    let resApi = await api.post('/activity/upvote', data);
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};

export const downVote = async (data) => {
  try {
    let resApi = await api.post('/activity/downvote', data);
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
    return error.response.data;
  }
};
