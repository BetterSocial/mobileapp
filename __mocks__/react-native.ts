import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN