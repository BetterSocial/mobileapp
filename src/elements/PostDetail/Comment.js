import * as React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import IconEn from 'react-native-vector-icons/Entypo';
import IconAnt from 'react-native-vector-icons/AntDesign';

import MemoIc_arrow_upvote_off from '../../assets/arrow/Ic_arrow_upvote_off';
import MemoIc_arrow_down_vote_off from '../../assets/arrow/Ic_arrow_down_vote_off';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

const Comment = ({username, comment, onPress, isLast = false}) => {
  return (
    <View
      style={[
        styles.container,
        {borderLeftColor: isLast ? '#fFF' : colors.gray1},
      ]}>
      <View style={styles.profile}>
        <Image
          source={require('../../assets/images/ProfileDefault.png')}
          style={styles.image}
        />
        <Text style={styles.username}>{username}</Text>
      </View>
      <Text style={styles.post}>{comment}</Text>
      <View style={styles.constainerFooter}>
        {isLast === true ? (
          <View style={{marginBottom: 8}} />
        ) : (
          <TouchableOpacity style={styles.btnReply} onPress={onPress}>
            <IconAnt name="back" size={15.77} color={colors.gray1} />
            <Text style={styles.btnReplyText}>Reply</Text>
          </TouchableOpacity>
        )}
        <TouchableOpacity style={styles.btnBlock}>
          <IconEn name="block" size={15.02} color={colors.gray1} />
        </TouchableOpacity>

        <TouchableOpacity style={styles.arrowup}>
          <MemoIc_arrow_down_vote_off width={18} height={18} />
        </TouchableOpacity>
        <TouchableOpacity>
          <MemoIc_arrow_upvote_off width={18} height={18} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Comment;

const styles = StyleSheet.create({
  image: {
    width: 24,
    height: 24,
  },
  container: {
    borderLeftWidth: 1,
  },
  username: {
    fontFamily: fonts.inter[600],
    fontSize: 12,
    color: '#828282',
    marginLeft: 17,
  },
  post: {
    fontFamily: fonts.inter[400],
    fontSize: 16,
    color: '#333333',
    marginLeft: 28,
  },
  profile: {
    flexDirection: 'row',
    marginLeft: -12,
  },
  constainerFooter: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginTop: 11.13,
  },
  btnReply: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  btnReplyText: {
    fontFamily: fonts.inter[400],
    fontSize: 13,
    color: '#C4C4C4',
    marginLeft: 8.98,
  },
  btnBlock: {
    marginLeft: 28.61,
    marginRight: 28.51,
  },
  arrowup: {
    marginRight: 33.04,
  },
});
