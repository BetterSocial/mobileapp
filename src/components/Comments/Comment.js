import * as React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import jwtDecode from 'jwt-decode';
import IconEn from 'react-native-vector-icons/Entypo';
import {useNavigation} from '@react-navigation/native';

import MemoIc_arrow_upvote_off from '../../assets/arrow/Ic_upvote_off';
import MemoIc_arrow_down_vote_off from '../../assets/arrow/Ic_downvote_off';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';
import {calculateTime} from '../../utils/time';
import MemoCommentReply from '../../assets/icon/CommentReply';
import {getAccessToken} from '../../utils/token';
import BlockUser from '../Blocking/BlockUser';
import ReportUser from '../Blocking/ReportUser';
import SpecificIssue from '../Blocking/SpecificIssue';
import {blockUser} from '../../service/blocking';
import Toast from 'react-native-simple-toast';
import JwtDecode from 'jwt-decode';

const Comment = ({
  user,
  comment,
  onPress,
  isLast = false,
  isLastInParent = false,
  time,
  style,
  photo,
  level,
  showLeftConnector = true,
  disableOnTextPress = false,
}) => {
  const navigation = useNavigation();
  const refBlockUser = React.useRef();
  const refSpecificIssue = React.useRef();
  const refReportUser = React.useRef();
  const [userId, setUserId] = React.useState('');
  const [username, setUsername] = React.useState('');
  const [postId, setPostId] = React.useState('');
  const [dataProfile, setDataProfile] = React.useState({});
  const [reportOption, setReportOption] = React.useState([]);
  const [messageReport, setMessageReport] = React.useState('');
  const [yourselfId, setYourselfId] = React.useState('');

  let onTextPress = () => {
    if (level >= 2 || disableOnTextPress) {
      return;
    }
    return onPress();
  };

  let openProfile = async () => {
    let selfAccessToken = await getAccessToken();
    let selfUserId = await jwtDecode(selfAccessToken).user_id;
    if (selfUserId === user.id) {
      return navigation.navigate('ProfileScreen');
    }
    return navigation.navigate('OtherProfile', {
      data: {
        user_id: selfUserId,
        other_id: user.id,
        username: user.data.username,
      },
    });
  };

  const setDataToState = (value) => {
    setUsername(value.user.data.username);
    setPostId(value.id);
    setUserId(value.user.id);
  };

  const onSkipOnlyBlock = () => {
    refReportUser.current.close();
    userBlock();
  };

  const onNextQuestion = (v) => {
    setReportOption(v);
    refReportUser.current.close();
    refSpecificIssue.current.open();
  };
  const onIssue = (v) => {
    refSpecificIssue.current.close();
    setMessageReport(v);
    setTimeout(() => {
      userBlock();
    }, 500);
  };

  const onSelectBlocking = (v) => {
    if (v !== 1) {
      refReportUser.current.open();
    } else {
      userBlock();
    }
    refBlockUser.current.close();
  };

  const userBlock = async () => {
    const data = {
      userId: userId,
      postId: postId,
      source: 'screen_post_detail',
      reason: reportOption,
      message: messageReport,
    };
    let result = await blockUser(data);
    if (result.code === 200) {
      Toast.show(
        'The user was blocked successfully. \nThanks for making BetterSocial better!',
        Toast.LONG,
      );
    } else {
      Toast.show('Your report was filed & will be investigated', Toast.LONG);
    }
    console.log('result block user ', result);
  };

  React.useEffect(() => {
    const parseToken = async () => {
      const value = await getAccessToken();
      if (value) {
        const decoded = await JwtDecode(value);
        setYourselfId(decoded.user_id);
      }
    };
    parseToken();
  }, []);

  return (
    <View
      style={styles.container({
        isLast,
        style,
        level,
        isLastInParent,
        showLeftConnector,
      })}>
      <TouchableOpacity onPress={openProfile}>
        <View style={styles.profile}>
          <Image
            source={
              photo
                ? {uri: photo}
                : require('../../assets/images/ProfileDefault.png')
            }
            style={styles.image}
          />
          <View style={styles.containerUsername}>
            <Text style={styles.username}>{user.data.username} â€¢</Text>
            <Text style={styles.time}> {calculateTime(time)}</Text>
          </View>
        </View>
      </TouchableOpacity>
      <TouchableOpacity onPress={onTextPress}>
        <Text style={styles.post}>{comment.data.text}</Text>
      </TouchableOpacity>
      <View style={styles.constainerFooter}>
        {isLast && level >= 2 ? (
          <View style={styles.gap} />
        ) : (
          <TouchableOpacity style={styles.btnReply} onPress={onPress}>
            <MemoCommentReply />
            <Text style={styles.btnReplyText}>Reply</Text>
          </TouchableOpacity>
        )}
        <TouchableOpacity
          style={[styles.btnBlock(comment.user.id === yourselfId), styles.btn]}
          onPress={() => {
            if (comment.user.id === yourselfId) {
              Toast.show("Can't Block yourself", Toast.LONG);
            } else {
              setDataToState(comment);
              refBlockUser.current.open();
            }
          }}>
          <IconEn name="block" size={15.02} color={colors.gray1} />
        </TouchableOpacity>

        <TouchableOpacity style={[styles.arrowup, styles.btn]}>
          <MemoIc_arrow_down_vote_off width={18} height={18} />
        </TouchableOpacity>
        <TouchableOpacity style={[styles.arrowdown, styles.btn]}>
          <MemoIc_arrow_upvote_off width={18} height={18} />
        </TouchableOpacity>
      </View>

      <BlockUser
        refBlockUser={refBlockUser}
        onSelect={(v) => onSelectBlocking(v)}
        username={username}
      />
      <ReportUser
        refReportUser={refReportUser}
        onSelect={onNextQuestion}
        onSkip={onSkipOnlyBlock}
      />
      <SpecificIssue
        refSpecificIssue={refSpecificIssue}
        onPress={onIssue}
        onSkip={onSkipOnlyBlock}
      />
    </View>
  );
};

export default Comment;

const styles = StyleSheet.create({
  btn: {
    // width: 30,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 24,
    height: 24,
    borderRadius: 12,
  },
  container: ({isLast, style, level, isLastInParent, showLeftConnector}) => ({
    width: '100%',
    borderLeftWidth: showLeftConnector ? 1 : 0,
    borderLeftColor: isLast
      ? level === 0
        ? colors.gray1
        : 'transparent'
      : colors.gray1,
    ...style,
  }),
  username: {
    fontFamily: fonts.inter[700],
    fontSize: 12,
    color: '#828282',
    lineHeight: 14,
    marginLeft: 16,
  },
  post: {
    fontFamily: fonts.inter[400],
    fontSize: 16,
    color: '#333333',
    marginLeft: 28,
  },
  profile: {
    flexDirection: 'row',
    marginLeft: -13,
  },
  constainerFooter: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginTop: 11.13,
    marginBottom: 12,
    marginLeft: 30,
  },
  btnReply: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
    flex: 1,
  },
  btnReplyText: {
    fontFamily: fonts.inter[400],
    fontSize: 13,
    color: '#C4C4C4',
    marginLeft: 8.98,
    marginRight: 14,
  },
  btnBlock: (isMySelf) => ({
    paddingHorizontal: 14,
    display: isMySelf ? 'none' : 'flex',
  }),
  arrowup: {
    paddingHorizontal: 14,
  },
  arrowdown: {
    paddingHorizontal: 14,
  },
  gap: {marginBottom: 8},
  time: {
    fontFamily: fonts.inter[400],
    fontSize: 10,
    color: '#828282',
    lineHeight: 12,
  },
  containerUsername: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});
