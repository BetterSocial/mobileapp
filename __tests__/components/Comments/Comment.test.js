import React, {useRef} from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import * as UserUtil from '../../../src/utils/users';
import * as serviceVote from '../../../src/service/vote';
import Comment, {isEqual} from '../../../src/components/Comments/Comment';

jest.mock('react-native/Libraries/Pressability/usePressability');
jest.mock('react-native/Libraries/Components/Pressable/Pressable');
const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({goBack: jest.fn(), navigate: mockNavigate}),
  useRoute: () => ({
    params: {}
  })
}));

const setState = jest.fn();
const openBlockComponent = jest.fn();
jest.mock('react', () => {
  const actualReact = jest.requireActual('react');

  return {
    ...actualReact,
    useState: jest.fn(),
    useRef: jest.fn()
  };
});

describe('Comment test should run correctly', () => {
  const user = {
    created_at: '2022-06-10T13:11:53.385703Z',
    data: {
      human_id: 'I4K3M10FGR78EWQQDNQ2',
      profile_pic_url:
        'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
      username: 'Agita'
    },
    id: 'c6c91b04-795c-404e-b012-ea28813a2006',
    updated_at: '2022-07-29T12:54:03.879150Z'
  };

  const comment = {
    activity_id: 'eeb07336-8fe8-11ed-b458-0e0d34fb440f',
    children_counts: {},
    data: {
      count_downvote: 0,
      count_upvote: 0,
      isNotSeen: true,
      text: 'from postman 1'
    },
    id: 'dd8e0ece-a27b-4356-9a37-0427afa86258',
    kind: 'comment',
    latest_children: {},
    parent: '',
    target_feeds: [
      'notification:f19ce509-e8ae-405f-91cf-ed19ce1ed96e',
      'notification:c6c91b04-795c-404e-b012-ea28813a2006'
    ],
    updated_at: '2023-01-09T06:57:52.892334Z',
    user: {
      created_at: '2022-06-10T13:11:53.385703Z',
      data: {
        human_id: 'I4K3M10FGR78EWQQDNQ2',
        profile_pic_url:
          'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
        username: 'Agita'
      },
      id: 'c6c91b04-795c-404e-b012-ea28813a2006',
      updated_at: '2022-07-29T12:54:03.879150Z'
    },
    user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
  };
  const photo =
    'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg';
  const level = 1;

  beforeEach(() => {
    jest.spyOn(React, 'useState').mockImplementation((initState) => [initState, setState]);
    useRef.mockReturnValueOnce({current: openBlockComponent});
  });

  it('Should match snapshot', () => {
    const onPress = jest.fn();
    const findComment = jest.fn();
    const updateVote = jest.fn();
    const time = '2023-01-09T14:13:31.890300Z';
    const {toJSON} = render(
      <Comment
        user={user}
        comment={comment}
        photo={photo}
        level={level}
        onPress={onPress}
        findCommentAndUpdate={findComment}
        updateVote={updateVote}
        time={time}
      />
    );
    expect(toJSON).toMatchSnapshot();
  });

  it('textPress fuction should run correctly', () => {
    const onPress = jest.fn();
    const findComment = jest.fn();
    const updateVote = jest.fn();
    const time = '2023-01-09T14:13:31.890300Z';
    const {getByTestId} = render(
      <Comment
        user={user}
        comment={comment}
        photo={photo}
        level={level}
        onPress={onPress}
        findCommentAndUpdate={findComment}
        updateVote={updateVote}
        time={time}
      />
    );
    fireEvent.press(getByTestId('textPress'));
    expect(onPress).toHaveBeenCalled();
    const {getByTestId: getByTestIdLevel2} = render(
      <Comment
        user={user}
        comment={comment}
        photo={photo}
        level={2}
        onPress={onPress}
        findCommentAndUpdate={findComment}
        updateVote={updateVote}
        time={time}
      />
    );
    fireEvent.press(getByTestIdLevel2('textPress'));
    expect(onPress).toHaveBeenCalled();
  });

  it('openProfile should run correctly', async () => {
    const onPress = jest.fn();
    const findComment = jest.fn();
    const updateVote = jest.fn();
    const time = '2023-01-09T14:13:31.890300Z';
    const spyGeUserId = jest.spyOn(UserUtil, 'getUserId');
    const {getByTestId} = render(
      <Comment
        user={user}
        comment={comment}
        photo={photo}
        level={level}
        onPress={onPress}
        findCommentAndUpdate={findComment}
        updateVote={updateVote}
        time={time}
      />
    );
    fireEvent.press(getByTestId('openProfile'));
    expect(spyGeUserId).toHaveBeenCalled();
  });

  it('btnDownvote should run correctly', () => {
    const onPress = jest.fn();
    const findComment = jest.fn();
    const updateVote = jest.fn();
    const time = '2023-01-09T14:13:31.890300Z';
    const serviceVoteMock = jest.spyOn(serviceVote, 'voteComment');
    const serviceIvoteMock = jest.spyOn(serviceVote, 'iVoteComment');
    const {getByTestId} = render(
      <Comment
        user={user}
        comment={comment}
        photo={photo}
        level={level}
        onPress={onPress}
        findCommentAndUpdate={findComment}
        updateVote={updateVote}
        time={time}
      />
    );
    fireEvent.press(getByTestId('btnDownvote'));
    expect(serviceVoteMock).toHaveBeenCalled();
    expect(serviceIvoteMock).toHaveBeenCalled();

    expect(setState).toHaveBeenCalled();
  });

  it('upvoteBtn should run correctly', () => {
    const onPress = jest.fn();
    const findComment = jest.fn();
    const updateVote = jest.fn();
    const time = '2023-01-09T14:13:31.890300Z';
    const serviceVoteMock = jest.spyOn(serviceVote, 'voteComment');
    const serviceIvoteMock = jest.spyOn(serviceVote, 'iVoteComment');
    const {getByTestId} = render(
      <Comment
        user={user}
        comment={comment}
        photo={photo}
        level={level}
        onPress={onPress}
        findCommentAndUpdate={findComment}
        updateVote={updateVote}
        time={time}
      />
    );
    fireEvent.press(getByTestId('upvoteBtn'));
    expect(serviceVoteMock).toHaveBeenCalled();
    expect(setState).toHaveBeenCalled();
    expect(serviceIvoteMock).toHaveBeenCalled();
  });

  it('react memo should should run correctly', async () => {
    expect(isEqual({comment: '1'}, {comment: '2'})).toBeFalsy();
    expect(isEqual({comment: '1'}, {comment: '1'})).toBeTruthy();
  });
});
