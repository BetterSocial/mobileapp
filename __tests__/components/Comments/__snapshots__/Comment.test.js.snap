// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment test should run correctly Should match snapshot 1`] = `[Function]`;
